src/Data/Semigroup/Traversable.hs

```diff
@@ -29,11 +32,35 @@ import Data.Semigroup
 #endif
 import Data.Semigroup.Traversable.Class
 import Data.Functor.Bind.Class
+import GHC.Generics (Rep1, Generic1, to1, from1)
 
 -- | Default implementation of 'foldMap1' given an implementation of 'Traversable1'.
 foldMap1Default :: (Traversable1 f, Semigroup m) => (a -> m) -> f a -> m
 foldMap1Default f = getConst . traverse1 (Const . f)
 
+-- | Generic implementation of 'traverse1'. Caveats:
+--
+--   1. Will not compile if @t@ is an empty constructor
+--   2. Will not compile if @t@ has some fields that don't mention @a@, for exmaple @data Bar a = MkBar a Int@
+--
+-- @since 5.3.8
+gtraverse1 ::
+  (Traversable1 (Rep1 t), Apply f, Generic1 t) =>
+  (a -> f b) ->
+  t a ->
+  f (t b)
+gtraverse1 f x = to1 <$> traverse1 f (from1 x)
+
+-- | Generic implementation of 'sequence1'
+--
+-- See 'gtraverse1' for some caveats.
```

Comment:
We have explicitly listed caveats elsewhere. Do you think we should spell them out in haddocks, or use links in all cases? I would prefer to be consistent.

Reply:
I don't have a strong opinion here. If you'd like, I would be fine with explicitly listing all the caveats for each function. Alternatively, if one function has exactly the same caveats as another nearby function, I would also be fine with pointing readers to the Haddocks of the other function.

Reply:
I have standardised on listing the caveats explicitly at most once in each module, and using links for the other functions in that module (if any).